crates/typst-library/src/foundations/calc.rs

```diff
@@ -1035,7 +1035,15 @@ pub fn quo(
     let divided = dividend
         .apply2(
             divisor.v,
-            |a, b| Some(DecNum::Int(a / b)),
+            |a, b| {
+                let q = a.checked_div(b)?;
+                // Round towards negative infinity if the fraction is negative.
+                Some(DecNum::Int(if (a < 0) != (b < 0) && a % b != 0 {
```

Comment:
Maybe this is a bit better?

```suggestion
                Some(DecNum::Int(if q < 0 && a % b != 0 {
```

By the way, this is the official implementation in the currently nightly-only Rust `std` function for this[^1]:

```rs
pub const fn div_floor(self, rhs: Self) -> Self {
    let d = self / rhs;
    let r = self % rhs;

    // If the remainder is non-zero, we need to subtract one if the
    // signs of self and rhs differ, as this means we rounded upwards
    // instead of downwards. We do this branchlessly by creating a mask
    // which is all-ones iff the signs differ, and 0 otherwise. Then by
    // adding this mask (which corresponds to the signed value -1), we
    // get our correction.
    let correction = (self ^ rhs) >> (Self::BITS - 1);
    if r != 0 {
        d + correction
    } else {
        d
    }
}
```

[^1]: https://doc.rust-lang.org/nightly/std/primitive.i64.html#method.div_floor

Reply:
I'm leaning towards keeping the current algorithm for the above reasons, curious to hear your thoughts

Reply:
Yeah let's keep it. Thanks for checking!